### PR TITLE
value in ParameterValidator is optional now

### PR DESCRIPTION
**Pull request by bartektartanus**
## Checklist before merge
- [ ] Related issue ID is placed at the beginning of PR title in \[brackets\] (can be GH issue or Nu Jira issue)
- [ ] Code is cleaned from temporary changes and commented out lines
- [ ] Parts of the code that are not easy to understand are documented in the code
- [ ] Changes are covered by automated tests
- [ ] Showcase in dev-application.conf added to demonstrate the feature
- [ ] Documentation added or updated
- [ ] Added entry in _Changelog.md_ describing the change from the perspective of a public distribution user
- [ ] Added _MigrationGuide.md_ entry in the appropriate subcategory if introducing a breaking change
- [ ] Verify that PR will be squashed during merge

### CODE REVIEW

File: components-api/src/main/scala/pl/touk/nussknacker/engine/api/definition/ParameterValidator.scala [hunk 1]
```diff
@@ -88,18 +112,16 @@ case object NotBlankParameterValidator extends ParameterValidator {
...
+  override def isValid(paramName: String, expression: Expression, value: Option[Any], label: Option[String])(
       implicit nodeId: NodeId
   ): Validated[PartSubGraphCompilationError, Unit] = {
     // FIXME: we should properly evaluate `possibleValues`
     val values = possibleValues.map(possibleValue => possibleValue.expression)
 
     value match {
-      case null                                        => valid(())
-      case s: String if values.contains(s"'$s'")       => valid(())
+      case None                                        => valid(())
```
[the remark this answers is not in the document]
Comment (bartektartanus) [reply components-api/src/main/scala/pl/touk/nussknacker/engine/api/definition/ParameterValidator.scala:122]: Yup, you're right. I've added test case for that and it fails :)

File: components-api/src/main/scala/pl/touk/nussknacker/engine/api/definition/ParameterValidator.scala [hunk 2]
```diff
@@ -66,15 +69,36 @@ case object NotNullParameterValidator extends ParameterValidator {
 
 }
 
+case object LiteralParameterValidator extends ParameterValidator {
```
[the remark this answers is not in the document]
Comment (bartektartanus) [reply components-api/src/main/scala/pl/touk/nussknacker/engine/api/definition/ParameterValidator.scala:72]: done

File: components-api/src/test/scala/pl/touk/nussknacker/engine/api/definition/ParameterValidatorSpec.scala [hunk 3]
```diff
@@ -70,62 +70,48 @@ class ParameterValidatorSpec extends AnyFunSuite with TableDrivenPropertyChecks
...
-    val validator = FixedValuesValidator(List(FixedExpressionValue("'a'", "a"), FixedExpressionValue("'b'", "b")))
-    forAll(
-      Table(
-        ("inputExpression", "isValid"),
-        (null, true),
-        ("", false),
-        ("  ", false),
-        ("\t", false),
-        (" \n ", false),
-        ("someString ", false),
```
[the remark this answers is not in the document]
Comment (bartektartanus) [reply components-api/src/test/scala/pl/touk/nussknacker/engine/api/definition/ParameterValidatorSpec.scala:88]: And they are still in repo :D

File: components-api/src/main/scala/pl/touk/nussknacker/engine/api/definition/ParameterValidator.scala [hunk 4]
```diff
@@ -88,53 +114,52 @@ case object NotBlankParameterValidator extends ParameterValidator {
...
-    value match {
-      case null                                        => valid(())
-      case s: String if values.contains(s"'$s'")       => valid(())
-      case _ if values.contains(expression.expression) => valid(())
-      case s: String => invalid(InvalidPropertyFixedValue(paramName, label, s"'$s'", possibleValues.map(_.expression)))
-      case x if values.contains(x.toString) => valid(())
-      case _ => invalid(InvalidPropertyFixedValue(paramName, label, value.toString, possibleValues.map(_.expression)))
+    // empty expression should not be validated - we want to chain validators
+    expression.expression match {
+      case ""                      => valid(())
```
Comment (arkadius) [inline components-api/src/main/scala/pl/touk/nussknacker/engine/api/definition/ParameterValidator.scala:125]: please check isBlank instead as we everywhere do

File: components-api/src/main/scala/pl/touk/nussknacker/engine/api/definition/ParameterValidator.scala [hunk 5]
```diff
@@ -88,53 +114,52 @@ case object NotBlankParameterValidator extends ParameterValidator {
...
+  // empty expression should not be validated - we want to chain validators
+  override def isValid(paramName: String, expression: Expression, value: Option[Any], label: Option[String])(
       implicit nodeId: NodeId
   ): Validated[PartSubGraphCompilationError, Unit] =
-    value match {
-      case null   => valid(())
-      case _: Int => valid(())
-      case _      => invalid(error(paramName, nodeId.id))
+    expression.expression match {
+      case ""                          => valid(())
```
Comment (arkadius) [inline components-api/src/main/scala/pl/touk/nussknacker/engine/api/definition/ParameterValidator.scala:160]: the same here

**Review by arkadius — APPROVED** [review]
Nice

File: components-api/src/main/scala/pl/touk/nussknacker/engine/api/definition/ParameterValidator.scala (hunk 5 above)
Comment (bartektartanus) [reply components-api/src/main/scala/pl/touk/nussknacker/engine/api/definition/ParameterValidator.scala]: fixed

**Pull request merged**